Stop referring to outdated milestone versions of Maven Surefire/Failsafe in the User Guide

Hi team, I noticed that tests were not executed anymore in older surefire versions (i.e. 2.22.2-3.0.0-M3), when upgrading to 5.12.0-M1.

For surefire 2.22.2 and 3.0.0-M3 we get this warning message, but in 3.0.0-M1 and 3.0.0-M2 the tests are silently ignored:
```
WARNING: TestEngine with ID 'junit-jupiter' failed to discover tests
org.junit.platform.commons.JUnitException: OutputDirectoryProvider not available; probably due to unaligned versions of the junit-platform-engine and junit-platform-launcher jars on the classpath/module path.
```

Was this an intentional change / does JUnit 5.12.0 drop support for those surefire versions?  

## Steps to reproduce

Run `./mvnw clean test` in the reproducer. For 5.11.4 the build fails, but for 5.12.0-M1 the build succeeds, because no tests are executed:

[reproducer.zip](https://github.com/user-attachments/files/18671741/reproducer.zip)

Is there a good reason to use old surefire/failsafe plugin versions instead of the most recent versions https://maven.apache.org/plugins/ At the moment surefire/failsafe version is 3.5.2 ? The version 2.22.2 is from 2019 while 3.0.0-M3 is from 2018... 
In the earlier versions the automatic identifcation to use a particluar engine... the JUnit-Jupiter-Engine has not been correct. Using the most recent version 3.5.2 will execute the tests correctly and fail as intended...

```
[INFO] --- compiler:3.13.0:testCompile (default-testCompile) @ maven-build-scan-quickstart ---
[INFO] Recompiling the module because of changed source code.
[INFO] Compiling 2 source files with javac [debug target 1.8] to target/test-classes
[WARNING] bootstrap class path not set in conjunction with -source 8
[INFO] 
[INFO] --- surefire:3.5.2:test (default-test) @ maven-build-scan-quickstart ---
[INFO] Using auto detected provider org.apache.maven.surefire.junitplatform.JUnitPlatformProvider
[INFO] 
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running example.TestB
[INFO] [stdout] testB1
[INFO] [stdout] testB2
[ERROR] Tests run: 2, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 0.027 s <<< FAILURE! -- in example.TestB
[ERROR] example.TestB.testB1 -- Time elapsed: 0.014 s <<< FAILURE!
org.opentest4j.AssertionFailedError: testB1 fail
	at org.junit.jupiter.api.AssertionUtils.fail(AssertionUtils.java:38)
	at org.junit.jupiter.api.Assertions.fail(Assertions.java:138)
	at example.TestB.testB1(TestB.java:8)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)

[INFO] Running example.TestA
[INFO] [stdout] testA1
[INFO] [stdout] testA2
[WARNING] Tests run: 3, Failures: 0, Errors: 0, Skipped: 1, Time elapsed: 0.003 s -- in example.TestA
[INFO] 
[INFO] Results:
[INFO] 
[ERROR] Failures: 
[ERROR]   TestB.testB1:8 testB1 fail
[INFO] 
[ERROR] Tests run: 5, Failures: 1, Errors: 0, Skipped: 1
[INFO] 
[INFO] --------------------------------------------------------------------------------------------------------------------------
[INFO] BUILD FAILURE
```

--- a/documentation/src/docs/asciidoc/user-guide/running-tests.adoc
+++ b/documentation/src/docs/asciidoc/user-guide/running-tests.adoc
@@ -254,24 +254,23 @@ additional dependency to the runtime classpath.
 [[running-tests-build-maven]]
 ==== Maven
 
-Starting with https://issues.apache.org/jira/browse/SUREFIRE-1330[version 2.22.0], Maven
-Surefire and Maven Failsafe provide
+Maven Surefire and Maven Failsafe provide
 https://maven.apache.org/surefire/maven-surefire-plugin/examples/junit-platform.html[native support]
 for executing tests on the JUnit Platform. The `pom.xml` file in the
 `{junit5-jupiter-starter-maven}` project demonstrates how to use the Maven Surefire plugin
 and can serve as a starting point for configuring your Maven build.
 
 [WARNING]
-.Use Maven Surefire/Failsafe 3.0.0-M4 or later to avoid interoperability issues
+.Use recent version of Maven Surefire/Failsafe to avoid interoperability issues
 ====
-Maven Surefire/Failsafe 3.0.0-M4
-https://issues.apache.org/jira/browse/SUREFIRE-1585[introduced support] for aligning the
-version of the JUnit Platform Launcher it uses with the JUnit Platform version found on
-the test runtime classpath. Therefore, it is recommended to use version 3.0.0-M4 or later
-to avoid interoperability issues.
+To avoid interoperability issues, it is recommended to use a recent versions of Maven
+Surefire/Failsafe (3.0.0 or later) because it automatically aligns the used version of the
+JUnit Platform Launcher with the JUnit Platform version found on the test runtime
+classpath.
 
-Alternatively, you can add a test dependency on the matching version of the JUnit Platform
-Launcher to your Maven build as follows.
+If you are using a version older than 3.0.0-M4, you can work around the missing alignment
+by adding a test dependency on the matching version of the JUnit Platform Launcher to your
+Maven build as follows.
 
 [source,xml]
 [subs=attributes+]

--- a/documentation/src/docs/asciidoc/user-guide/running-tests.adoc
+++ b/documentation/src/docs/asciidoc/user-guide/running-tests.adoc
@@ -263,10 +263,10 @@ and can serve as a starting point for configuring your Maven build.
 [WARNING]
 .Use recent version of Maven Surefire/Failsafe to avoid interoperability issues
 ====
-To avoid interoperability issues, it is recommended to use a recent versions of Maven
-Surefire/Failsafe (3.0.0 or later) because it automatically aligns the used version of the
-JUnit Platform Launcher with the JUnit Platform version found on the test runtime
-classpath.
+To avoid interoperability issues, it is recommended to use a recent version of Maven
+Surefire/Failsafe (3.0.0 or later) because it automatically aligns the version of the
+JUnit Platform Launcher that is used with the JUnit Platform version found on the test
+runtime classpath.
 
 If you are using a version older than 3.0.0-M4, you can work around the missing alignment
 by adding a test dependency on the matching version of the JUnit Platform Launcher to your

